File: LeetCode/Math/Golang/Leetcode793-preimage-size-of-factorial-zeroes-function.go
package sbgo

func preimageSizeFZF(k int) int {
	f := func(x int) int {
		res := 0
		for x > 0 {
			res += x / 5
			x /= 5
		}
		return res
	}

	calcu := func(k int) int {
		l, r := -1, int(1e18)
		for l < r {
			mid := (l + r + 1) >> 1
			if f(mid) <= k {
				l = mid
			} else {
				r = mid - 1
			}
		}
		return r
	}

	return calcu(k) - calcu(k-1)
}

File: LeetCode/Math/JS/Leetcode793-preimage-size-of-factorial-zeroes-function.js
var preimageSizeFZF = function(k) {
    let calcu = function(k) {
        let [l,r] = [-1,1e18];
        while (l < r) {
            const mid = Math.floor((l + r + 1) / 2);
            if(f(mid) <= k) l = mid;
            else r = mid - 1;
        }
        return r;
    }

    let f = function(x) {
        let res = 0;
        while(x != 0) {
            res += Math.floor(x / 5);
            x = Math.floor(x / 5);
        }
        return res;
    }
    return calcu(k) - calcu(k - 1);
};